lib/code_corps/stripe_service/webhook_processing/webhook_processor.ex

```diff
@@ -5,79 +5,104 @@ defmodule CodeCorps.StripeService.WebhookProcessing.WebhookProcessor do
-      case handler.handle_event(json) |> Tuple.to_list do
-        [:ok, :unhandled_event] -> event |> set_unhandled
-        [:ok | _results]        -> event |> set_processed
-        [:error | _error]       -> event |> set_errored
-      end
+  defp do_process(id, user_id, handler, json) do
+    with {:ok, %Stripe.Event{id: api_event_id, type: api_event_type, user_id: api_user_id}} <- retrieve_event_from_api(id, user_id),
+         {:ok, endpoint} <- infer_endpoint_from_handler(handler),
+         {:ok, %StripeEvent{} = event} <- find_or_create_event(api_event_id, api_event_type, api_user_id, endpoint)
+    do
+      handle_event(json, event, handler)
     else
       {:error, :already_processing} -> nil
     end
   end
 
-  defp find_or_create_event(id_from_stripe, type) do
+  defp event_matches_environment?(livemode) do
+    case Application.get_env(:code_corps, :stripe_env) do
+      :prod -> livemode
+      _ -> !livemode
+    end
+  end
+
+  defp find_or_create_event(id_from_stripe, type, user_id, endpoint) do
     case find_event(id_from_stripe) do
       %StripeEvent{status: "processing"} -> {:error, :already_processing}
       %StripeEvent{} = event -> {:ok, event}
-      nil -> create_event(id_from_stripe, type)
+      nil -> create_event(id_from_stripe, endpoint, type, user_id)
     end
   end
 
   defp find_event(id_from_stripe) do
     Repo.get_by(StripeEvent, id_from_stripe: id_from_stripe)
   end
 
-  defp create_event(id_from_stripe, type) do
-    %StripeEvent{} |> StripeEvent.create_changeset(%{id_from_stripe: id_from_stripe, type: type}) |> Repo.insert
+  defp handle_event(json, event, handler) do
+    case handler.handle_event(json) |> Tuple.to_list do
+      [:ok, :unhandled_event] -> event |> set_unhandled
+      [:ok | _results]        -> event |> set_processed
+      [:error | _error]       -> event |> set_errored
+    end
   end
 
-  defp set_processed(%StripeEvent{} = event) do
-    event |> StripeEvent.update_changeset(%{status: "processed"}) |> Repo.update
+  defp infer_endpoint_from_handler(ConnectEventHandler), do: {:ok, "connect"}
+  defp infer_endpoint_from_handler(PlatformEventHandler), do: {:ok, "platform"}
+  defp infer_endpoint_from_handler(_), do: {:error, :invalid_handler}
+
+  defp retrieve_event_from_api(id, nil), do: @api.Event.retrieve(id)
+  defp retrieve_event_from_api(id, user_id), do: @api.Event.retrieve(id, connect_account: user_id)
+
+  defp create_event(id_from_stripe, endpoint, type, user_id) do
+    %StripeEvent{} |> StripeEvent.create_changeset(%{endpoint: endpoint, id_from_stripe: id_from_stripe, type: type, user_id: user_id}) |> Repo.insert
   end
 
   defp set_errored(%StripeEvent{} = event) do
     event |> StripeEvent.update_changeset(%{status: "errored"}) |> Repo.update
   end
 
+  defp set_processed(%StripeEvent{} = event) do
+    event |> StripeEvent.update_changeset(%{status: "processed"}) |> Repo.update
+  end
```

Comment:
Yeah, I wasn't sure about this, hemmed and hawed, but ultimately this is probably okay. Was thinking we could just have done `set_status/2` with a string as a second argument.